pkg/selector/selector.go

```diff
@@ -201,20 +212,139 @@ func (itf Selector) FilterInstanceTypes(filters Filters) ([]*instancetypes.Detai
 		filteredInstanceTypes = append(filteredInstanceTypes, it)
 	}
 
-	return sortInstanceTypeInfo(filteredInstanceTypes), nil
+	return filteredInstanceTypes, nil
+}
+
+// isSortFilterValid determines whether the given sortFilter matches any
+// of the given valid flags
+func isSortFilterValid(sortFilter string, validFlags ...string) bool {
+	for _, flag := range validFlags {
+		if sortFilter == flag {
+			return true
+		}
+	}
+
+	return false
 }
 
-// sortInstanceTypeInfo will sort based on instance type info alpha-numerically
-func sortInstanceTypeInfo(instanceTypeInfoSlice []*instancetypes.Details) []*instancetypes.Details {
-	if len(instanceTypeInfoSlice) < 2 {
-		return instanceTypeInfoSlice
+// SortInstanceTypes accepts a list of instance type details, a sort filter flag, and a sort direction flag and
+// returns a list of instance type details sorted based on the sort filter and sort direction.
+// Sort filter options: "on-demand-price", "spot-price", "vcpu", "memory", "instance-type-name".
+// Sort direction options: "ascending", "descending".
+func (itf Selector) SortInstanceTypes(instanceTypes []*instancetypes.Details, sortFilter *string, sortDirection *string) ([]*instancetypes.Details, error) {
+	if len(instanceTypes) <= 1 {
+		return instanceTypes, nil
+	}
+
+	// validate direction flag and determine sorting direction
+	if sortDirection == nil {
+		return nil, fmt.Errorf("sort direction is nil")
+	}
+	var isDescending bool
+	switch *sortDirection {
+	case sortDescending:
+		isDescending = true
+	case sortAscending:
+		isDescending = false
+	default:
+		return nil, fmt.Errorf("invalid sort direction: %s", *sortDirection)
+	}
+
+	// validate filter flag
+	if sortFilter == nil {
+		return nil, fmt.Errorf("sort filter is nil")
+	} else if !isSortFilterValid(*sortFilter, sortODPrice, sortSpotPrice, sortVCPU, sortMemory, sortName) {
```

Comment:
Since we will always be checking against the same set of flags, we should move the expected strings into the function itself instead of passing in, e.g., something like:
```
func isSortFilterValid(sortFilter *string) bool {
...
	for _, flag := range []string{sortODPrice, sortSpotPrice, sortVCPU, sortMemory, sortName} {
		if *sortFilter == flag {
			return true
		}
	}
        return false
}
```

Reply:
That is a very good point. I have refactored `isSortFilterValid` into `validateSortFilter` which only takes in the sort filter flag.